li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -461,6 +481,48 @@ private long positionMain(TopicPartition partition, Duration timeout) {
     }
   }
 
+  private ConsumerRecordsProcessingException handleRecordProcessingException(Collection<TopicPartition> topicPartitions) {
+    if (_lastProcessedResult == null || !_lastProcessedResult.hasException()) {
+      return null;
+    }
+
+    ConsumerRecordsProcessingException crpe = null;
+    if (topicPartitions == null || topicPartitions.isEmpty()) {
+      // seek past offset for all topic-partitions that hit an exception
+      _lastProcessedResult.offsets().forEach((tp, o) -> _kafkaConsumer.seek(tp, o.getResumeOffset()));
+      crpe = _lastProcessedResult.exception();
+    } else {
+      // seek past offset for topic-partition in the collection that hit an exception
+      if (_lastProcessedResult.hasError(topicPartitions)) {
+        Map<TopicPartition, ConsumerRecordsProcessResult<K, V>.OffsetPair> offsets = _lastProcessedResult.offsets();
+        topicPartitions.forEach(tp -> {
+          if (offsets.containsKey(tp)) {
+            _kafkaConsumer.seek(tp, offsets.get(tp).getResumeOffset());
+          }
+        });
+        crpe = _lastProcessedResult.exception(topicPartitions);
+      }
+
+      // if topic-partitions don't have an exception then just drop cached exceptions and move-on
+    }
+    _lastProcessedResult = null;
+    return crpe;
+  }
+
+  private void seekToCurrentOffsetsOnRecordProcessingExceptions() {
+    // seek to offset which had an exception
+    if (_lastProcessedResult != null && _lastProcessedResult.hasException()) {
+      _lastProcessedResult.offsets().forEach((k, v) -> _kafkaConsumer.seek(k, v.getCurrentOffset()));
+    }
+  }
+
+  private void clearRecordProcessingException() {
+    if (_lastProcessedResult != null && _lastProcessedResult.hasException()) {
+      LOG.trace("Clearing all Record Processing Exceptions");
```

Comment:
Can this be logged at warn or info level.  This would have been an exception delivered to a user.  Trace is almost as bad as just swallowing the exception.

Reply:
And can you log the exceptions that are being dropped.